src/roi/Roi.ts

```diff
@@ -626,3 +654,103 @@ function getBorders(roi: Roi): Border[] {
     };
   });
 }
+
+function getEllipse(roi: Roi, scale: number): Ellipse {
+  const nbSD = 2;
+
+  let xCenter = roi.centroid.column;
+  let yCenter = roi.centroid.row;
+
+  let xCentered = roi.points.map((point: number[]) => point[0] - xCenter);
+  let yCentered = roi.points.map((point: number[]) => point[1] - yCenter);
+
+  let centeredXVariance = xVariance(xCentered, { unbiased: false });
+  let centeredYVariance = xVariance(yCentered, { unbiased: false });
+
+  let centeredCovariance = xyCovariance(
+    {
+      x: xCentered,
+      y: yCentered,
+    },
+    { unbiased: false },
+  );
+
+  //spectral decomposition of the sample covariance matrix
+  let sampleCovarianceMatrix = [
+    [centeredXVariance, centeredCovariance],
+    [centeredCovariance, centeredYVariance],
+  ];
+  let e = new EigenvalueDecomposition(sampleCovarianceMatrix);
+  let eigenvalues = e.realEigenvalues;
+  let vectors = e.eigenvectorMatrix;
+
+  let radiusMajor: number;
+  let radiusMinor: number;
+  let vectorMajor: number[];
+  let vectorMinor: number[];
+
+  if (eigenvalues[0] > eigenvalues[1]) {
```

Comment:
Is it possible to add a test that covers this condition?

Reply:
> Is it possible to add a test that covers this condition?

I don't think it is possible to create a test that covers it because the eigenvalue array is sorted at the end of its computation. So it is impossible for element 0 to be bigger than element 1.

Reply:
If you're confident that the condition is impossible, you can remove it and add `assert(eigenvalues[0] <= eigenvalues[1]);` instead (be careful to import assert from https://github.com/image-js/image-js-typescript/blob/main/src/utils/assert.ts.